File: rust/examples/linux/pressure/bmp085/minimal/src/main.rs
use linux_embedded_hal::I2cdev;
use periph::chips::pressure::Bmp085Minimal;

fn main() {
    let i2c_bus: u8 = std::env::var("I2C_BUS").ok().and_then(|v| v.parse().ok()).unwrap_or(1);

    let dev = I2cdev::new(format!("/dev/i2c-{}", i2c_bus)).expect("open i2c bus");
    let mut bmp = Bmp085Minimal::new(dev, 0x77).expect("init BMP085"); // Create BMP085 driver, (connection)

    for _ in 0..5 {
        let t = bmp.temperature().expect("read temperature");      // Read temperature, () → f32 C
        let p = bmp.pressure().expect("read pressure");            // Read pressure, () → f32 Pa
        println!("{:.1} C, {:.1} Pa", t, p);
        std::thread::sleep(std::time::Duration::from_secs(1));
    }
}

File: cpp/examples/linux/pressure/BMP085/minimal/main.cpp
#ifndef TEST_I2C_BUS
#define TEST_I2C_BUS 1
#endif
#ifndef TEST_ADDR
#define TEST_ADDR 0x77
#endif

#include <cstdio>
#include "I2CConnectionLinux.h"
#include "BMP085.h"

int main() {
    I2CConnectionLinux connection(TEST_I2C_BUS, TEST_ADDR);
    BMP085Minimal bmp(connection);                          // Create BMP085 driver, (connection)

    for (int i = 0; i < 5; i++) {
        float t = bmp.temperature();                        // Read temperature, () → float C
        float p = bmp.pressure();                          // Read pressure, () → float Pa
        printf("%.1f C, %.1f Pa\n", t, p);
    }

    printf("===DONE: 0 passed, 0 failed===\n");
    return 0;
}
